File: elixir/space-age/lib/space_age.ex
defmodule SpaceAge do
  @type planet ::
          :mercury
          | :venus
          | :earth
          | :mars
          | :jupiter
          | :saturn
          | :uranus
          | :neptune

  @doc """
  Return the number of years a person that has lived for 'seconds' seconds is
  aged on 'planet'.
  """
  @spec age_on(planet, pos_integer) :: float
  def age_on(planet, seconds) do
    earth_age = seconds / 31_557_600

    case planet do
      :mercury -> earth_age / 0.2408467
      :venus -> earth_age / 0.61519726
      :earth -> earth_age
      :mars -> earth_age / 1.8808158
      :jupiter -> earth_age / 11.862615
      :saturn -> earth_age / 29.447498
      :uranus -> earth_age / 84.016846
      :neptune -> earth_age / 164.79132
    end
  end
end

File: fsharp/space-age/SpaceAge.fs
module SpaceAge

type Planet =
    | Mercury   
    | Venus
    | Earth
    | Mars
    | Jupiter
    | Saturn
    | Uranus
    | Neptune

let age (planet: Planet) (seconds: int64) : float =
    let earthYearInSeconds = 31_557_600.
    let earthOrbitalYears = float seconds / earthYearInSeconds

    let earthYearsInAOrbit =
        match planet with
        | Mercury -> 0.2408467
        | Venus -> 0.61519726
        | Earth -> 1.0
        | Mars -> 1.8808158
        | Jupiter -> 11.862615
        | Saturn -> 29.447498
        | Uranus -> 84.016846
        | Neptune -> 164.79132

    earthOrbitalYears / earthYearsInAOrbit